Sources/TuistKit/Services/BuildService.swift

```diff
@@ -70,41 +77,87 @@ final class BuildService {
+                try buildScheme(
+                    scheme: $0,
+                    graphTraverser: graphTraverser,
+                    workspacePath: workspacePath,
+                    clean: !cleaned && clean,
+                    configuration: configuration
+                )
                 cleaned = true
             }
         }
 
+        if let productsPath = productsPath {
+            try copyBuildProducts(to: productsPath, workspacePath: workspacePath)
+        }
+
         logger.log(level: .notice, "The project built successfully", metadata: .success)
     }
 
     // MARK: - private
 
-    private func buildScheme(scheme: Scheme, graphTraverser: GraphTraversing, path: AbsolutePath, clean: Bool, configuration: String?) throws {
+    private func buildScheme(
+        scheme: Scheme,
+        graphTraverser: GraphTraversing,
+        workspacePath: AbsolutePath,
+        clean: Bool,
+        configuration: String?
+    ) throws {
         logger.log(level: .notice, "Building scheme \(scheme.name)", metadata: .section)
         guard let (project, target) = buildGraphInspector.buildableTarget(scheme: scheme, graphTraverser: graphTraverser) else {
             throw BuildServiceError.schemeWithoutBuildableTargets(scheme: scheme.name)
         }
-        let workspacePath = try buildGraphInspector.workspacePath(directory: path)!
+
         let buildArguments = buildGraphInspector.buildArguments(project: project, target: target, configuration: configuration, skipSigning: false)
-        _ = try xcodebuildController.build(
-            .workspace(workspacePath),
-            scheme: scheme.name,
-            clean: clean,
-            arguments: buildArguments
-        )
-        .printFormattedOutput()
-        .toBlocking()
-        .last()
+
+        _ = try xcodeBuildController
+            .build(
+                .workspace(workspacePath),
+                scheme: scheme.name,
+                clean: clean,
+                arguments: buildArguments
+            )
+            .printFormattedOutput()
+            .toBlocking()
+            .last()
+    }
+
+    private func copyBuildProducts(
+        to outputPath: AbsolutePath,
+        workspacePath: AbsolutePath
+    ) throws {
+        let buildOutputDirectory = try xcodeBuildLocator.locateProductsDirectory(for: workspacePath)
+
+        guard FileHandler.shared.exists(buildOutputDirectory) else {
+            logger.error("No build products found at \(buildOutputDirectory.pathString), skipping copy products")
+            return
+        }
+
+        if FileHandler.shared.exists(outputPath) {
+            try FileHandler.shared.delete(outputPath)
+        }
+
+        logger.log(level: .notice, "Copying build products to \(outputPath.pathString)", metadata: .subsection)
+        try FileHandler.shared.copy(from: buildOutputDirectory, to: outputPath)
```

Comment:
Am I right assuming that we might copy files from previous builds? Here are some ideas:
- We can get a timestamp before building, and recursively copy the files whose updated at date is more recent than the timestamp.
- Since we build a configuration, you can copy he directory for only that configuration.

Also, as a user, let's say I have the following under the `--products-path` directory:

```
products/
  Debug/
```

If I build for `Release`, I'd expect to end up with something like this:

```
products/
  Debug/
  Release/
```
Note that we are not overriding previous configurations. This behavior is similar to the SPM's.

Reply:
> Am I right assuming that we might copy files from previous builds?

We delete the directory for the current build configuration so in this example if the path was `.build` and building `Release` configuration, the implementation right now deletes the directory `.build/Release` and then copies from derived data to the cleaned `.build/Release`.

> Note that we are not overriding previous configurations. This behavior is similar to the SPM's.

For sure! We do are doing this here as well since we copy to a directory based on configuration. So `--build-path .build --configuration Release` -> `.build/Release` getting copied

Reply:
Checkout comments below to see if we should update this to also include the scheme name in the products directory.